### PR TITLE
Geo: better handling of malformed geo_points

### LINKED ISSUE #35419: geo_point parsing doesn't ignore malformed points represented as a string
Looks like there are some use cases in #16137 that were not fully addressed in #16833. In particular, some values in string presentation of a geo_point are not handled correctly. To reproduce:

```
DELETE example

PUT example
{
    "mappings": {
        "doc": {
            "properties": {
                "location": {
                    "type": "geo_point",
                    "ignore_malformed": true
                }
            }
        }
    }
}

PUT example/doc/1
{
  "location": "NaN,NaN"
}

PUT example/doc/2
{
  "location": "-,-"
}
```

### PR DESCRIPTION
**Pull request by imotov**
Improves handling of malformed geo_points when `ignore_malformed` is
set to true

Closes #35419

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java [hunk 1]
```diff
@@ -260,7 +260,11 @@ protected void parse(ParseContext context, GeoPoint point) throws IOException {
                 throw new IllegalArgumentException("illegal longitude value [" + point.lon() + "] for " + name());
             }
         } else {
-            GeoUtils.normalizePoint(point);
+            if (isNormalizable(point.lat()) && isNormalizable(point.lat())) {
```
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java]: should one of these be point.lon()?

Comment (cbuescher) [reply server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java]: If so, there should probably also be a test added that would have caught this.

**Review by cbuescher — CHANGES_REQUESTED** [review]
@imotov looks good to me except for a small issue I left a comment on.

**Comment by imotov** [comment]
@cbuescher I pushed the changes that you have requested. Could you take another look?

**Review by cbuescher — APPROVED** [review]
Thanks for the change @imotov, LGTM

**Pull request merged**